### ISSUE TITLE
Can't pickle local object

### ISSUE THREAD

**Issue by vibber** [epitaque/dreambooth_depth2img#4]
Fetching 15 files: 100%|█████████████████████████████████████████████████████████████| 15/15 [00:00<00:00, 1674.60it/s]
For each image in C:\Users\VERTIGO\Documents\my_concept, creating a depth image.
100%|███████████████████████████████████████████████████████████████████████████████| 26/26 [00:00<00:00, 13035.13it/s]
For each image in C:\Users\VERTIGO\Documents\classdata, creating a depth image.
100%|█████████████████████████████████████████████████████████████████████████████| 200/200 [00:00<00:00, 22282.27it/s]
Steps:   0%|                                                                                  | 0/2000 [00:00<?, ?it/s]╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ <string>:1 in <module>                                                                           │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\multiprocessing\spawn.py:116 in spawn_main   │
│                                                                                                  │
│   113 │   │   resource_tracker._resource_tracker._fd = tracker_fd                                │
│   114 │   │   fd = pipe_handle                                                                   │
│   115 │   │   parent_sentinel = os.dup(pipe_handle)                                              │
│ ❱ 116 │   exitcode = _main(fd, parent_sentinel)                                                  │
│   117 │   sys.exit(exitcode)                                                                     │
│   118                                                                                            │
│   119                                                                                            │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\multiprocessing\spawn.py:126 in _main        │
│                                                                                                  │
│   123 │   │   try:                                                                               │
│   124 │   │   │   preparation_data = reduction.pickle.load(from_parent)                          │
│   125 │   │   │   prepare(preparation_data)                                                      │
│ ❱ 126 │   │   │   self = reduction.pickle.load(from_parent)                                      │
│   127 │   │   finally:                                                                           │
│   128 │   │   │   del process.current_process()._inheriting                                      │
│   129 │   return self._bootstrap(parent_sentinel)                                                │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
EOFError: Ran out of input
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Users\VERTIGO\dreamboothdepth\dreambooth_depth2img\train_dreambooth.py:837 in <module>        │
│                                                                                                  │
│   834                                                                                            │
│   835 if __name__ == "__main__":                                                                 │
│   836 │   args = parse_args()                                                                    │
│ ❱ 837 │   main(args)                                                                             │
│   838                                                                                            │
│                                                                                                  │
│ C:\Users\VERTIGO\dreamboothdepth\dreambooth_depth2img\train_dreambooth.py:731 in main            │
│                                                                                                  │
│   728 │   │   unet.train()                                                                       │
│   729 │   │   if args.train_text_encoder:                                                        │
│   730 │   │   │   text_encoder.train()                                                           │
│ ❱ 731 │   │   for step, batch in enumerate(train_dataloader):                                    │
│   732 │   │   │   # Skip steps until we reach the resumed step                                   │
│   733 │   │   │   if args.resume_from_checkpoint and epoch == first_epoch and step < resume_st   │
│   734 │   │   │   │   if step % args.gradient_accumulation_steps == 0:                           │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\site-packages\accelerate\data_loader.py:372  │
│ in __iter__                                                                                      │
│                                                                                                  │
│   369 │   │   with suppress(Exception):                                                          │
│   370 │   │   │   length = getattr(self.dataset, "total_dataset_length", len(self.dataset))      │
│   371 │   │   │   self.gradient_state._set_remainder(length % self.total_batch_size)             │
│ ❱ 372 │   │   dataloader_iter = super().__iter__()                                               │
│   373 │   │   # We iterate one batch ahead to check when we are at the end                       │
│   374 │   │   try:                                                                               │
│   375 │   │   │   current_batch = next(dataloader_iter)                                          │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\site-packages\torch\utils\data\dataloader.py │
│ :444 in __iter__                                                                                 │
│                                                                                                  │
│    441 │   │   │   │   self._iterator._reset(self)                                               │
│    442 │   │   │   return self._iterator                                                         │
│    443 │   │   else:                                                                             │
│ ❱  444 │   │   │   return self._get_iterator()                                                   │
│    445 │                                                                                         │
│    446 │   @property                                                                             │
│    447 │   def _auto_collation(self):                                                            │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\site-packages\torch\utils\data\dataloader.py │
│ :390 in _get_iterator                                                                            │
│                                                                                                  │
│    387 │   │   │   return _SingleProcessDataLoaderIter(self)                                     │
│    388 │   │   else:                                                                             │
│    389 │   │   │   self.check_worker_number_rationality()                                        │
│ ❱  390 │   │   │   return _MultiProcessingDataLoaderIter(self)                                   │
│    391 │                                                                                         │
│    392 │   @property                                                                             │
│    393 │   def multiprocessing_context(self):                                                    │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\site-packages\torch\utils\data\dataloader.py │
│ :1077 in __init__                                                                                │
│                                                                                                  │
│   1074 │   │   │   #     it started, so that we do not call .join() if program dies              │
│   1075 │   │   │   #     before it starts, and __del__ tries to join but will get:               │
│   1076 │   │   │   #     AssertionError: can only join a started process.                        │
│ ❱ 1077 │   │   │   w.start()                                                                     │
│   1078 │   │   │   self._index_queues.append(index_queue)                                        │
│   1079 │   │   │   self._workers.append(w)                                                       │
│   1080                                                                                           │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\multiprocessing\process.py:121 in start      │
│                                                                                                  │
│   118 │   │   assert not _current_process._config.get('daemon'), \                               │
│   119 │   │   │      'daemonic processes are not allowed to have children'                       │
│   120 │   │   _cleanup()                                                                         │
│ ❱ 121 │   │   self._popen = self._Popen(self)                                                    │
│   122 │   │   self._sentinel = self._popen.sentinel                                              │
│   123 │   │   # Avoid a refcycle if the target function holds an indirect                        │
│   124 │   │   # reference to the process object (see bpo-30775)                                  │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\multiprocessing\context.py:224 in _Popen     │
│                                                                                                  │
│   221 │   _start_method = None                                                                   │
│   222 │   @staticmethod                                                                          │
│   223 │   def _Popen(process_obj):                                                               │
│ ❱ 224 │   │   return _default_context.get_context().Process._Popen(process_obj)                  │
│   225                                                                                            │
│   226 class DefaultContext(BaseContext):                                                         │
│   227 │   Process = Process                                                                      │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\multiprocessing\context.py:327 in _Popen     │
│                                                                                                  │
│   324 │   │   @staticmethod                                                                      │
│   325 │   │   def _Popen(process_obj):                                                           │
│   326 │   │   │   from .popen_spawn_win32 import Popen                                           │
│ ❱ 327 │   │   │   return Popen(process_obj)                                                      │
│   328 │                                                                                          │
│   329 │   class SpawnContext(BaseContext):                                                       │
│   330 │   │   _name = 'spawn'                                                                    │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\multiprocessing\popen_spawn_win32.py:93 in   │
│ __init__                                                                                         │
│                                                                                                  │
│    90 │   │   │   set_spawning_popen(self)                                                       │
│    91 │   │   │   try:                                                                           │
│    92 │   │   │   │   reduction.dump(prep_data, to_child)                                        │
│ ❱  93 │   │   │   │   reduction.dump(process_obj, to_child)                                      │
│    94 │   │   │   finally:                                                                       │
│    95 │   │   │   │   set_spawning_popen(None)                                                   │
│    96                                                                                            │
│                                                                                                  │
│ c:\users\vertigo\anaconda3\envs\dreamboothdepth\lib\multiprocessing\reduction.py:60 in dump      │
│                                                                                                  │
│    57                                                                                            │
│    58 def dump(obj, file, protocol=None):                                                        │
│    59 │   '''Replacement for pickle.dump() using ForkingPickler.'''                              │
│ ❱  60 │   ForkingPickler(file, protocol).dump(obj)                                               │
│    61                                                                                            │
│    62 #                                                                                          │
│    63 # Platform specific definitions                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: Can't pickle local object 'main.<locals>.<lambda>'

**Comment by vibber** [comment]
`Can't pickle local object 'main.<locals>.<lambda>'`

**Comment by vibber** [comment]
accelerate test runs fine btw

**Comment by epitaque** [comment]
Out of curiosity, are you running on multiple gpus?

**Comment by vibber** [comment]
no. I also tried without accelerate. But I've found out I have the same issue with main dream booth script from diffusers, so I've also asked there

**Comment by thibaudart** [comment]
On windows, you need to remove the line 640:

        num_workers=1,
